packages/api/src/webhooks/cannon.ts

```
@@ -52,10 +52,73 @@ export default class WebhookCannon {
 
   async start() {
     console.log("WEBHOOK CANNON STARTED");
-    await this.queue.consume(this.handleQueueMsg.bind(this));
+    await this.queue.consume("webhooks", this.handleWebhookQueue.bind(this));
+    await this.queue.consume("events", this.handleEventsQueue.bind(this));
   }
 
-  async handleQueueMsg(data: ConsumeMessage) {
+  async handleEventsQueue (data: ConsumeMessage) {
+    let message;
+    try {
+      message = JSON.parse(data.content.toString());
+      console.log("events: got message", message);
+    } catch (err) {
+      console.log("events: error parsing message", err);
+      this.queue.ack(data);
+      return;
+    }
+    let event = message;
+
+    try {
+      const { data: webhooksList } = await this.db.webhook.listSubscribed(
+        event.userId,
+        event.event
+      );
+  
+      console.log("webhooks : ", webhooksList);
+      let stream = await this.db.stream.get(event.streamId);
+      if (!stream) {
+        // if stream isn't found. don't fire the webhook, log an error
+        return console.error(
```

We need to throw an error instead of returning from here as well